### PR TITLE
ItemProperties: Add IngredientProperty

### PR DESCRIPTION
**Pull request by kristofbolyai**
Artemis now supports ingredient highlights.

image

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java [hunk 1]
```diff
@@ -71,6 +71,30 @@ public class ItemHighlightFeature extends UserFeature {
     @Config
     public static CustomColor craftedHighlightColor = new CustomColor(0, 138, 138);
 
+    @Config
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java:74]: Does it really make sense to have highlight enabled per tier? I think a single boolean to enable/disable ingredient highlighting is better.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java:74]: Wynntils has a minimum tier for highlight, but I guess I can see this getting used. Not saying we need this 100%, but also not like it is even close to heavy or anything.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java:74]: I'm just thinking it is tricky for the user to disable ingredient highlighting, they need to disable four switches.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java:74]: Ah, I can understand that. I am not sure, I will think about this. I wouldn't do overall toggle PLUS per tier toggles thought, that is for sure.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/inventory/ItemHighlightFeature.java:74]: Whatever, it's not really important. I seriously doubt anyone will ever disable this. It's almost one of the raison d'etre of Wynntils...

**Review by magicus — DISMISSED** [review]

File: common/src/main/java/com/wynntils/wc/custom/item/properties/IngredientProperty.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+public class IngredientProperty extends ItemProperty implements HighlightProperty {
+
+    private final IngredientTier tier;
+
+    public IngredientProperty(WynnItemStack item) {
+        super(item);
+
+        String name = WynnUtils.normalizeBadString(item.getHoverName().getString());
+
+        if (name.endsWith(ChatFormatting.GOLD + " [" + ChatFormatting.YELLOW + "✫" + ChatFormatting.DARK_GRAY + "✫✫"
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/wc/custom/item/properties/IngredientProperty.java:27]: This could be exported to a separate method.

**Review by magicus — APPROVED** [review]

File: common/src/main/java/com/wynntils/wc/custom/item/properties/IngredientProperty.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    public boolean isInventoryHighlight() {
+        return true;
+    }
+
+    @Override
+    public boolean isHotbarHighlight() {
+        return false;
+    }
+
+    public enum IngredientTier {
```
Comment (P0keDev) [inline common/src/main/java/com/wynntils/wc/custom/item/properties/IngredientProperty.java:71]: once we add support for the ingredient api this enum should be expanded to mirror the ItemTier one, i.e. have the tier colors & toggles defined within them. for now i'm fine with the current implementation

**Review by P0keDev — APPROVED** [review]

**Pull request merged**